cocotb/share/lib/verilator/verilator.cpp

```diff
@@ -0,0 +1,85 @@
+#include "Vtop.h"
+#include "verilated.h"
+#include "verilated_vpi.h"
+
+#if VM_TRACE
+# include <verilated_vcd_c.h>
+#endif
+
+vluint64_t main_time = 0;       // Current simulation time
+
+double sc_time_stamp() {       // Called by $time in Verilog
+    return main_time;           // converts to double, to match
+                                // what SystemC does
+}
+
+extern "C" {
+void vlog_startup_routines_bootstrap(void);
+}
+
+int main(int argc, char** argv) {
+    Verilated::commandArgs(argc, argv);
+#ifdef VERILATOR_SIM_DEBUG
+    Verilated::debug(99);
+#endif
+    Vtop* top = new Vtop("");
+    Verilated::fatalOnVpiError(false); // otherwise it will fail on systemtf
+
+    vlog_startup_routines_bootstrap();
+    VerilatedVpi::callCbs(cbStartOfSimulation);
+
+#if VM_TRACE
+    Verilated::traceEverOn(true);
+
+    VerilatedVcdC *tfp = new VerilatedVcdC;
```

Comment:
Tempting to wait until we get #1137 in, then use `unique_ptr` here

Reply:
no need to wait for it, the Verilator build can handle it